Cython 3.0 is still in beta so we do not plan to package it yet.
Since Cython >= 3.0 is needed for Python 3.10+ only we could safely relax
the required Cython version.

--- gevent-22.10.2/pyproject.toml.orig
+++ gevent-22.10.2/pyproject.toml
@@ -17,7 +17,7 @@
      # This was fixed in 3.0a5 (https://github.com/cython/cython/issues/3578)
      # 3.0a6 fixes an issue cythonizing source on 32-bit platforms.
      # 3.0a9 is needed for Python 3.10.
-     "Cython >= 3.0a11",
+     "Cython >= 0.29.14",
      # See version requirements in setup.py
      "cffi >= 1.12.3 ; platform_python_implementation == 'CPython'",
      # Python 3.7 requires at least 0.4.14, which is ABI incompatible with earlier
